service/collector_test.go

```diff
@@ -66,9 +69,27 @@ func TestCollector_StartAsGoRoutine(t *testing.T) {
 	collectorTelemetry = &colTelemetry{}
 	defer func() { collectorTelemetry = preservedAppTelemetry }()
 
-	factories, err := defaultcomponents.Components()
+	receivers, err := component.MakeReceiverFactoryMap(
+		otlpreceiver.NewFactory(),
+	)
 	require.NoError(t, err)
 
+	processors, err := component.MakeProcessorFactoryMap(
+		batchprocessor.NewFactory(),
+	)
+	require.NoError(t, err)
+
+	exporters, err := component.MakeExporterFactoryMap(
+		otlpexporter.NewFactory(),
+	)
+	require.NoError(t, err)
+
+	factories := component.Factories{
+		Receivers:  receivers,
+		Processors: processors,
+		Exporters:  exporters,
+	}
```

Comment:
This doesn't include extensions but the other tests do.  All of the places where `component.Factories` are created look very similar.  Should this be extracted into a function that can be reused?

Reply:
Thank you. I have extracted the components from testdata/otel-config.yaml into a function in `example_factories.go` for the tests that use this file, in addition to `TestCollector_StartAsGoRoutine` which is associated to the configured components of `configStr` in `collector_test.go` file.